src/userVerifier.ts

```diff
@@ -1,17 +1,21 @@
-import { IAddress, IVerifiable, IVerifier } from "./interface";
+import { IVerifiable } from "./interface";
 import { UserPublicKey } from "./userKeys";
 
+interface IAddress {
+  pubkey(): Buffer;
+}
+
 /**
  * ed25519 signature verification
  */
-export class UserVerifier implements IVerifier {
-
+export class UserVerifier {
   publicKey: UserPublicKey;
+
   constructor(publicKey: UserPublicKey) {
     this.publicKey = publicKey;
   }
 
-  static fromAddress(address: IAddress): IVerifier {
+  static fromAddress(address: IAddress): UserVerifier {
```

Comment:
We should receive simple abstract interfaces as input, but it's OK to return concrete, complete objects.